frappe/public/js/frappe/views/workspace/blocks/paragraph.js

```diff
@@ -27,62 +27,95 @@ export default class Paragraph extends Block {
 	}
 
 	onKeyUp(e) {
+		if (!this.wrapper) return;
+		this.show_hide_block_list(true);
 		if (e.code !== 'Backspace' && e.code !== 'Delete') {
 			return;
 		}
 
 		const {textContent} = this._element;
 
 		if (textContent === '') {
+			this.show_hide_block_list();
 			this._element.innerHTML = '';
 		}
 	}
 
+	show_hide_block_list(hide) {
+		let $wrapper = $(this.wrapper).hasClass('ce-paragraph') ? $(this.wrapper.parentElement) : $(this.wrapper);
+		let $block_list_container = $wrapper.find('.block-list-container.dropdown-list');
+		$block_list_container.removeClass('hidden');
+		hide && $block_list_container.addClass('hidden');
+	}
+
 	drawView() {
 		let div = document.createElement('DIV');
 
 		div.classList.add(this._CSS.wrapper, this._CSS.block, 'widget');
 		div.contentEditable = false;
-		div.dataset.placeholder = this.api.i18n.t(this._placeholder);
 
 		if (!this.readOnly) {
 			div.contentEditable = true;
+			div.addEventListener('focus', () => {
+				const {textContent} = this._element;
+				if (textContent !== '') return;
+				this.show_hide_block_list();
+			});
+			div.addEventListener('blur', () => {
+				setTimeout(() => this.show_hide_block_list(true), 10);
+			});
+			div.dataset.placeholder = this.api.i18n.t(this._placeholder);
 			div.addEventListener('keyup', this.onKeyUp);
 		}
 		return div;
 	}
 
+	open_block_list() {
+		let dropdown_title = 'Templates';
+		let $block_list_container = $(`
+			<div class="block-list-container dropdown-list">
+				<div class="dropdown-title">${dropdown_title.toUpperCase()}</div>
+			</div>
+		`);
+
+		let all_blocks = frappe.wspace_block.blocks;
```

Comment:
@shariquerik please remove `wspace` and replace with `workspace` everywhere. As a rule, we don't use abbreviations or short forms as it creates entropy for new people wanting to contribute also kills the readability of the code.